packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -447,6 +452,10 @@ class DatePicker extends React.Component {
       selectedDateInPicker = this.state.selectedDate;
     }
 
+    if (this.isDefaultDateAcceptable && !this.isDateWithinRange(selectedDateInPicker)) {
+      selectedDateInPicker = undefined;
```

Comment:
Should we do this only for uncontrolled component? I would think if consumers are using a controlled component, we should just let them manage the value instead of clearing it.

Reply:
I'm not sure, basically the date-time-picker is using a controlled version of date-picker, and the issue we were having was that when we assigned a value to date-picker in date-time and it was out of range, `selectedDateinPicker` would still be the set value. If that value was wildly inaccurate and not within the same year of the accepted range it cause issues with the calender dropdown where the only way for the user to get to the accepted range was to click the left arrow to manually go through months until you got there.

Reply:
We can keep it this way. I just think that as a controlled DatePicker and from a DatePicker perspective, it shouldn't update the value passed in since it should allow the higher order component manage all its state and value.